client-misk-spanner/src/main/kotlin/app/cash/backfila/client/misk/spanner/internal/SpannerBackend.kt

```diff
@@ -0,0 +1,78 @@
+package app.cash.backfila.client.misk.spanner.internal
+
+import app.cash.backfila.client.DeleteBy
+import app.cash.backfila.client.Description
+import app.cash.backfila.client.misk.spanner.ForSpannerBackend
+import app.cash.backfila.client.misk.spanner.SpannerBackfill
+import app.cash.backfila.client.parseDeleteByDate
+import app.cash.backfila.client.spi.BackfillBackend
+import app.cash.backfila.client.spi.BackfillOperator
+import app.cash.backfila.client.spi.BackfilaParametersOperator
+import app.cash.backfila.client.spi.BackfillRegistration
+import com.google.cloud.spanner.Spanner
+import com.google.inject.Injector
+import com.google.inject.TypeLiteral
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import java.lang.reflect.ParameterizedType
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.findAnnotation
+
+@Singleton
+internal class SpannerBackend @Inject constructor(
```

Comment:
General observation: this looks _almost_ entirely like something that could be made generic across backfills.